lib/fastlane/plugin/wpmreleasetoolkit/helper/git_helper.rb

```diff
@@ -123,22 +123,14 @@ def self.list_local_tags(matching: '*')
       # @param [Array<String>] tag_names The list of tags to delete
       # @param [Bool] delete_on_remote If true, will also delete the tag from the remote. Otherwise, it will only be deleted locally.
       #
-      def self.delete_tags(tag_names, delete_on_remote: false)
-
-        tag_names = [tag_names] unless tag_names.is_a? Array
+      def self.delete_tags(tags_to_delete, delete_on_remote: false)
 
         g = Git.open(Dir.pwd)
-        local_tag_names = g.tags.map do |tag|
-          tag.name
-        end
-
-        tag_names.each do |tag|
-          Action.sh('git', 'tag', '-d', *tag) if local_tag_names.include? tag
-        end
+        local_tag_names = g.tags.map(&:name)
 
-        if delete_on_remote
-          remote_refs = tag_names.map { |tag| ":refs/tags/#{tag}" }
-          Action.sh('git', 'push', 'origin', *remote_refs)
+        Array(tags_to_delete).select { |tag| local_tag_names.include? tag }.each do |tag|
+          g.delete_tag(tag)
+          g.push('origin', ":refs/tags/#{tag}") if delete_on_remote 
         end
```

Comment:
⚠️  That will only delete remote tags if they are also present locally though, because you call the `g.push` on the filtered / `.select`'d list. So if I am not the one who created the intermediate tags and didn't do a `git pull --tags --force` for a while to fetch those tags on my local machine, then the remote tags won't get deleted when I run it on my machine…

```suggestion
        Array(tags_to_delete).each do |tag|
          g.delete_tag(tag) if local_tag_names.include?(tag)
          g.push('origin', ":refs/tags/#{tag}") if delete_on_remote 
        end
```

Reply:
I'm not sure, but it looks like[ currently `tags_to_delete`](https://github.com/wordpress-mobile/release-toolkit/blob/d3d0d673deebb9a288050efa0da9e92bdb279403/lib/fastlane/plugin/wpmreleasetoolkit/actions/ios/ios_clear_intermediate_tags.rb#L20) are already [read locally](https://github.com/wordpress-mobile/release-toolkit/blob/d3d0d673deebb9a288050efa0da9e92bdb279403/lib/fastlane/plugin/wpmreleasetoolkit/actions/ios/ios_clear_intermediate_tags.rb#L16), right? 🤔

Reply:
True (that was just not yet the case in the state the PR was in when I reviewed and wrote that, but Jeremy fixed it since).

Tho tbh I find the code suggested above easier to read than the one with `.select`… but maybe that's just me 🤷